src/android/com/microsoftopentechnologies/intellij/helpers/azure/sdk/AzureSDKManagerImpl.java

```diff
@@ -380,73 +368,457 @@ public void deleteVirtualMachine(@NotNull VirtualMachine vm, boolean deleteFromS
     }
 
     @NotNull
-    private static List<VirtualMachine> processDeploymentSlot(@NotNull ComputeManagementClient client, @NotNull String subscriptionId, @NotNull String serviceName,
-                                                              @NotNull DeploymentSlot slot, @NotNull List<VirtualMachine> vmList)
-            throws Exception {
-        DeploymentGetResponse deployment = getDeployment(client, serviceName, slot);
+    @Override
+    public List<StorageAccount> getStorageAccounts(@NotNull String subscriptionId) throws AzureCmdException {
+        List<StorageAccount> saList = new ArrayList<StorageAccount>();
+        StorageManagementClient client = null;
 
-        if (deployment == null) {
-            return vmList;
-        }
+        try {
+            client = getStorageManagementClient(subscriptionId);
 
-        for (Role role : deployment.getRoles()) {
-            if (role.getRoleType() != null
-                    && role.getRoleType().equals(PERSISTENT_VM_ROLE)) {
-                VirtualMachine vm = new VirtualMachine(
-                        role.getRoleName() != null ? role.getRoleName() : "",
-                        serviceName,
-                        deployment.getUri() != null ? deployment.getUri().toString() : "",
-                        slot.toString(),
-                        deployment.getName() != null ? deployment.getName() : "",
-                        role.getRoleSize() != null ? role.getRoleSize() : "",
-                        deployment.getStatus() != null ? deployment.getStatus().toString() : "",
+            ArrayList<com.microsoft.windowsazure.management.storage.models.StorageAccount> storageAccounts =
+                    getStorageAccounts(client).getStorageAccounts();
+
+            if (storageAccounts == null) {
+                return saList;
+            }
+
+            for (com.microsoft.windowsazure.management.storage.models.StorageAccount storageAccount : storageAccounts) {
+                String key = "";
+
+                if (storageAccount.getName() != null) {
+                    String primaryKey = getStorageAccountKeys(client, storageAccount.getName()).getPrimaryKey();
```

Comment:
This call will result in an I/O operation. This is also an opportunity for running the query for all the storage accounts in parallel.